jni/db_bin.c

```diff
@@ -251,6 +259,10 @@ void add_search(int hb, int vb, int he, int ve) {
       }
       while (r<ni && I[r].b <= e) {
         if (I[r].e >= b) {
+          if (n >= MAX_RES) {
+            // Disallow writing out of bounds.
+            break;
```

Comment:
```suggestion
            return;
```

There's no need to continue with the search if we are over bound.